deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml

```diff
@@ -6,12 +7,15 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: {{ template "sumologic.metadata.name.opentelemetry.operator.instrumentation.job" . }}
+  name: {{ template "sumologic.metadata.name.opentelemetry.operator.instrumentation.job" . }}-{{ include (print $.Template.BasePath "/opentelemetry-operator/configmap-instrumentation-cr.yaml") $ctx | sha256sum | substr 0 5 }}
```

Comment:
Why is this necessary?

Reply:
It's necessary for a case when customer want's to add additional namespace to be instrumented. For `helm upgrade...` existing job will be not triggered. There're two ways to solve this:
* delete job - I don't want to do that, in case of issues we will lose information about failure reason
* create new job with different name - it makes sure that job will be triggered again and operator will take care of creation of instrumentation resource

Reply:
> For helm upgrade... existing job will be not triggered. 

Our Chart setup Job does get triggered every time. So why not this one?

Reply:
> > For helm upgrade... existing job will be not triggered.
> 
> Our Chart setup Job does get triggered every time. So why not this one?

Because we remove setup job when it finish

Reply:
Can we do the same thing here?

Reply:
We are going to lose information in case of failed instrumentation resource creation. It will make harder for us to investigate/troubleshoot issues.

Reply:
Can't we just use the default hook delete policy from Helm: https://helm.sh/docs/topics/charts_hooks/#hook-deletion-policies? Then it's only going to be deleted after you try installing again, but that sounds like it should work if you want logs from failed executions.

The reason I'm asking about this is that adding the hash to the name feels like an antipattern in K8s and I'd rather avoid it.

Reply:
Hmm. Will that work for non-helm installation?

Reply:
Does this work for setup in the non-Helm installation? If so, how?

Reply:
> Does this work for setup in the non-Helm installation

Good question. I don't know the answer :fearful: